avocado/utils/partition.py

```diff
@@ -298,18 +300,27 @@ def unmount(self, force=True):
         """
         with MtabLock():
             mountpoint = self.get_mountpoint()
```

Comment:
One more catch here: if any test uses `mount()` method with a different mountpoint (optional arg `mountpoint`) then we fail to get that mountpoint here and hence there is a chance unmount will fail. If that is the case we should either restrict taking an alternate mountpoint in mount method or have an optional `mountpoint` arg specified here in `unmount()` method.

Is that fair or am I missing something?